File: src/exercises/dot_dsl/python/template.py
NODE = 0
EDGE = 1
ATTR = 2

class Node:
    def __init__(self, name: str, attrs: dict = None):
        self.name = name
        self.attrs = attrs or {}

class Edge:
    def __init__(self, src: str, dst: str, attrs: dict = None):
        self.src = src
        self.dst = dst
        self.attrs = attrs or {}

class Graph:
    def __init__(self, data=None):
        self.nodes = []
        self.edges = []
        self.attrs = {}
        # Your code here

File: src/exercises/dot_dsl/go/template.go
package main

type Node struct {
	Name  string
	Attrs map[string]string
}

type Edge struct {
	From  string
	To    string
	Attrs map[string]string
}

type Graph struct {
	Nodes []Node
	Edges []Edge
	Attrs map[string]string
}

func NewGraph() *Graph {
	return &Graph{
		Nodes: []Node{},
		Edges: []Edge{},
		Attrs: make(map[string]string),
	}
}
